### PR TITLE
Connect to the Redux

### PR DESCRIPTION
**Pull request by akhror-valiev**
##Summary 
- Used Provider to connect Redux store.
- Used useSelector consume global state.
- Used useDispatch to import add and remove actions.

### CODE REVIEW

File: src/components/Book/Book.js [hunk 1]
```diff
@@ -1,26 +1,35 @@
+/* eslint-disable react/button-has-type */
```
Comment (CollinsTatang) [inline src/components/Book/Book.js]: - [ ] Please do not disable your `eslint linters check` remove this and make sure your linters pass positively

File: src/components/Book/Book.js [hunk 2]
```diff
@@ -1,26 +1,35 @@
...
-        {' '}
-        % Completed
-      </p>
-      <p>Current Chapter</p>
-      <p>{chapter}</p>
-      <button type="button">Update progress</button>
-    </li>
-  </div>
-);
+// eslint-disable-next-line react/prop-types
```
Comment (CollinsTatang) [inline src/components/Book/Book.js]: - [ ] Please do not disable your `eslint linters check` remove this and make sure your linters pass positively

File: src/components/Book/Book.js [hunk 3]
```diff
@@ -1,26 +1,35 @@
...
+const Book = ({ id, title, author }) => {
+  const dispatch = useDispatch();
+
+  const handleRemoveClick = () => {
+    // eslint-disable-next-line object-shorthand
```
Comment (CollinsTatang) [inline src/components/Book/Book.js]: - [ ] Please do not disable your `eslint linters check` remove this and make sure your linters pass positively

File: src/pages/Books/Books.js [hunk 4]
```diff
@@ -1,40 +1,25 @@
...
-  author: 'person2',
-  status: '50%',
-  chapter: 2,
+const Books = () => {
+  const books = useSelector((state) => state.books);
+  return (
+    <div className="container">
+      {
+        books.map((book) => (
+          // eslint-disable-next-line react/jsx-key
```
Comment (CollinsTatang) [inline src/pages/Books/Books.js]: - [ ] Please do not disable your `eslint linters check` remove this and make sure your linters pass positively

File: src/redux/books/books.js [hunk 5]
```diff
@@ -13,9 +16,16 @@ const bookSlice = createSlice({
...
         id: Date.now(),
         title: action.payload.title,
+        author: action.payload.author,
       };
       state.push(newBook);
     },
+    togleComplete: (state, action) => {
+      const index = state.findIndex((book) => book.id === action.payload.id);
+
+      // eslint-disable-next-line no-param-reassign
```
Comment (CollinsTatang) [inline src/redux/books/books.js]: - [ ] Please do not disable your `eslint linters check` remove this and make sure your linters pass positively

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @akhror-valiev, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.

- [ ] Please the `Check Status` Button should display `Under Construction` when a user clicks.
underCon


_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Book/Book.js (hunk 1 above)
Comment (medaminedev66) [reply src/components/Book/Book.js]: This linter issue can be solved. Kindly add the type attribute to your button.

File: src/components/Book/Book.js (hunk 2 above)
Comment (medaminedev66) [reply src/components/Book/Book.js]: This linter issue can be solved. Kindly create the prop-types for your component.

File: src/components/Book/Book.js (hunk 3 above)
Comment (medaminedev66) [reply src/components/Book/Book.js]: This linter issue can be solved. Kindly take a look at [this article](https://eslint.org/docs/latest/rules/object-shorthand) to learn how to solve it.

File: src/pages/Books/Books.js (hunk 4 above)
Comment (medaminedev66) [reply src/pages/Books/Books.js]: This linter issue can be solved. Kindly take a look at [this article](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-key.md) to learn how to solve it.

File: src/redux/books/books.js (hunk 5 above)
Comment (medaminedev66) [reply src/redux/books/books.js]: Since Redux enforce us to resign params, you can disable this linter rule in this case.

**Review by medaminedev66 — COMMENTED** [review]
Hello @akhror-valiev 👋,

_This is an additional review as a result of the second opinion request._

As developers, we usually do not disable linters rules, linters help us to write better code, find typos, and syntax errors, avoid using undeclared variables, avoid calls to undefined or deprecated functions, and can help us to fix the errors faster, and reduce bugs before execution.

Kindly take check out the comments I left below the previous review.

Keep rocking 🚀!

**Review by m05e5 — APPROVED** [review]
Hi @akhror-valiev  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights
✔️ Descriptive PR 
✔️ your linters are all ok

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**